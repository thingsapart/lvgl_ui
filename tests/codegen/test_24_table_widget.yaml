# A simple 2x3 table with custom column widths and populated cells.
- type: table
  size: [240, 100]
  center: []
  lv_table_set_column_width: [0, 60]
  lv_table_set_cell_value: [0, 0, 'Header 1']
  lv_table_set_cell_value: [0, 1, 'Header 2']
  lv_table_set_cell_value: [1, 0, 'Cell 1']
  lv_table_set_cell_value: [1, 1, 'Cell 2']
