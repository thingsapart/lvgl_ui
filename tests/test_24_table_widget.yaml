# A simple 2x3 table with custom column widths and populated cells.
- type: table
  size: [240, 100]
  center: []
  col_width: [60]
  set_cell_value: [0, 0, 'Header 1']
  set_cell_value: [0, 1, 'Header 2']
  set_cell_value: [1, 0, 'Cell 1']
  set_cell_value: [1, 1, 'Cell 2']
