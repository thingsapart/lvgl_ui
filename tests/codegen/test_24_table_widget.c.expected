/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: table_0 (table)
    lv_obj_t* table_0 = lv_table_create(parent);

    lv_obj_set_style_size(table_0, 240, 100, 0);
    lv_obj_center(table_0);
    lv_table_set_column_width(table_0, 0, 60);
    lv_table_set_cell_value(table_0, 0, 0, "Header 1");
    lv_table_set_cell_value(table_0, 0, 1, "Header 2");
    lv_table_set_cell_value(table_0, 1, 0, "Cell 1");
    lv_table_set_cell_value(table_0, 1, 1, "Cell 2");

}
